bin/setup

```diff
@@ -1,8 +1,79 @@
 #!/usr/bin/env bash
-set -euo pipefail
-IFS=$'\n\t'
-set -vx
 
-bundle install
+# bundle install
```

Comment:
🙀  hahahahah

whata setup!

Reply:
@fernandes don't you like it???? :joy: 
Well it's probably my second bash script ever....

Reply:
hahahah not about like or not, was thinking how incredible is the setup, a lot of gems, on specific directories hahaha

I didn't read it carefully, if you'd like I can try to provide you an alternative version

Reply:
I don't think is really important to have a super cool script there....but if you time to do that why not?

Reply:
ahhhhh super cool is not the main goal, was thinking just about future maintenance heheh

just some DRY

after years working as sysadmin, shell script is part of my toolbox heheh

Reply:
Makes totally sense! @apotonick was talking about having something that allow us to have a 2.2 easily as well :+1:

Reply:
```bash
# Lib
function message() {
  echo "${GREEN}$* ${WHITE}"
}

function unpack_gem() {
  [ ! -d ${UNPACK_DIR} ] && mkdir -p ${UNPACK_DIR}
  cd ${UNPACK_DIR} # I don't like to cd in scripts but I cd to root in the end
  wget -q "https://github.com/trailblazer/${1}/archive/${2}.zip"
  [ $? -ne 0 ] && echo "Problem downloading ${1} ${2}"
  unzip -q "${2}.zip"
  mv "${1}-${2#v}" "${1}"
  rm "${2}.zip"
  cd ${BASEDIR}
}

function remove() {
  rm -rf "${BASEDIR}/${1}"
}

function copy_gem() {
  [ ! -d ${TRB_DEST} ] && mkdir -p ${TRB_DEST}
  cp -R "${BASEDIR}/unpack/${1}/lib/trailblazer/." ${TRB_DEST}
}

function process_gem() {
  message "Setting up gem ${1}"
  unpack_gem "${1}" "${2}"
  copy_gem "${1}" ${TRB_DEST}
}

# Setup
GREEN="\033[0;32m"
WHITE="\033[0;37m"
SCRIPTDIR="$( cd "$(dirname "$0")" ; pwd -P )"
BASEDIR=`dirname ${SCRIPTDIR}`
UNPACK_DIR="${BASEDIR}/unpack"
LIB_DIR="${BASEDIR}/lib/trailblazer"


# Just need to configure both variables
TRB_DEST="${LIB_DIR}/v2_1"
TRB_VERSION="2.1"

# Running!
message "We are going to setup trailblazer version ${TRB_VERSION}"
message "On your folder ${BASEDIR}"
message "Shall we continue?"

select yn in "Yes" "No"; do
  case $yn in
    Yes ) echo 'Initializing setup'; break;;
    No ) echo 'Bye!'; exit;;
  esac
done

message "Getting folders ready"
rm -rf ${TRB_DEST}
rm -rf ${UNPACK_DIR}

# Configure the gems version here
message "Setting up gem trailblazer"
unpack_gem "trailblazer" "v2.1.0.beta5"
copy_gem "trailblazer" ${TRB_DEST}
cp "${UNPACK_DIR}/trailblazer/lib/trailblazer.rb" ${TRB_DEST}

process_gem "trailblazer-operation" "v0.2.4"
process_gem "trailblazer-activity" "v0.5.1"
process_gem "trailblazer-macro" "v2.1.0.beta3"
process_gem "trailblazer-macro-contract" "v2.1.0.beta2"
process_gem "trailblazer-context" "v0.1.2"
process_gem "trailblazer-rails" "v2.1.1"

# Teardown
rm -rf ${UNPACK_DIR}
```

gimme your thougts...

Reply:
I avoid to `cd` and use relative folders because this already gave A LOT of headache hahahaha

that's why the setup is a little bit tedious, but when running the script, we just handle absolute URLs (specially because we do a lot of rm -rf)

wonder you doing something like this in your /home/user/src/trailblazer-future 3 times:

```bash
git clone repo_url
cd repo_dir
cp file dest
cd ..
rm -rf dir
```

then your connection if offline... what happens?

in `/home/user/src/trailblazer-future` 1st time
1. git clone fails
1. cd fails because no folder to join
1. cp fails
1. cd .. goes one level up, now in `/home/user/src`
1. rm -rf drops your repo project

2nd time... now in `/home/user/src`

1. git clone fails
1. cd fails because no folder to join
1. cp fails
1. cd .. goes one level up, now in `/home/user`
1. rm -rf drops your src dir (hope you have backup)

3rd time... no in `/home/user` I think you can wonder what's gonna happen:

1. git clone fails
1. cd fails because no folder to join
1. cp fails
1. cd .. goes one level up, now in `/home`
1. rm -rf drops your home dir 💥 

that's why using relative directories is so dangerous 😉 

before you ask, no... it never happened to me, a friend tell me what happened to him :trollface:  hahahahha ok, was a test server 😜

Reply:
it looks good to me! @apotonick @seuros what do you guys think?

Reply:
@fernandes thanks for the effort but @seuros has implement a different script in the `meta` repo :+1:

Reply:
no worries 😉